### ISSUE TITLE
Testing with Eureka

### ISSUE THREAD

**Issue by micheal-swiggs** [spring-cloud/spring-cloud-netflix#906]
I'm just wondering if there are any examples of how to test Eureka based feign clients for a microservice architecture? I have tests that make calls to other services. At the moment these break as I get `java.lang.RuntimeException: com.netflix.client.ClientException: Load balancer does not have available server for client: service-name`. Do I need to use `eureka.client.enabled: false` ?

**Comment by spencergibb** [comment]
Usually when we do testing with ribbon, we swap out the server list implementation (we essentially remove eureka from ribbon).  See [here](https://github.com/spring-cloud/spring-cloud-netflix/blob/3994a3e439bb66a282cc2feb4989fd18123f2b6c/spring-cloud-netflix-core/src/test/java/org/springframework/cloud/netflix/feign/valid/scanning/FeignClientScanningTests.java#L113-L125) for a full example.

TL;DR

Ribbon configuration class:
```
	// Load balancer with fixed server list for "local" pointing to localhost
	public static class LocalRibbonClientConfiguration {
		@Value("${local.server.port}")
		private int port = 0;

		@Bean
		public ServerList<Server> ribbonServerList() {
			return new StaticServerList<>(new Server("localhost", this.port));
		}
	}
```

Annotate your test configuration class like so:
```
@RibbonClient(name = "localapp", configuration = LocalRibbonClientConfiguration.class)
```

Then calls to the `localapp` via `RestTemplate` or `LoadBalancerClient` will get `localhost:<testport>`.

**Comment by micheal-swiggs** [comment]
Thanks will give this a shot.
